tools/witx/tests/wasi.rs

```diff
@@ -69,8 +72,90 @@ fn document_wasi_snapshot() {
     use witx::Documentation;
     println!(
         "{}",
-        witx::load(&["../../phases/snapshot/witx/wasi_snapshot_preview1.witx"])
+        witx::load(&witx::phases::snapshot().unwrap())
             .unwrap_or_else(|e| panic!("failed to parse: {}", e))
             .to_md()
     );
 }
+
+fn dos2unix(s: &str) -> String {
+    let mut t = String::new();
+    t.reserve(s.len());
+    for c in s.chars() {
+        if c != '\r' {
+            t.push(c)
+        }
+    }
+    t
+}
+
+fn diff_against_filesystem(expected: &str, path: &Path) {
+    let actual =
+        fs::read_to_string(path).unwrap_or_else(|e| panic!("couldn't read {:?}: {:?}", path, e));
```

Comment:
FWIW, it possible to render `Path` and `PathBuf` to `fmt::Display` using [`Path::display(&self)`] function. Then again, this is not a must, but renders `"some/path"` instead of `Path { ... }` as is the case with `fmt::Debug` for `Path` and `PathBuf` :-)

[`Path::display(&self)`]: https://doc.rust-lang.org/std/path/struct.Path.html#method.display

Reply:
TIL!

Reply:
TIL!